packages/ckeditor5-typing/src/twostepcaretmovement.ts

```diff
@@ -563,6 +570,71 @@ export default class TwoStepCaretMovement extends Plugin {
 		} );
 	}
 
+	/**
+	 * Enables special handling for middle-link touches on iOS devices.
+	 * When a user touches the middle part of a link (not on edges), the cursor
+	 * will be positioned at the end of the link to allow easy typing after the link.
+	 *
+	 * This is iOS-specific behavior to improve the user experience when working with links.
+	 *
+	 * See: https://github.com/ckeditor/ckeditor5/issues/18023
+	 */
+	private _enableMiddleLinkTouchHandlerForIOS(): void {
+		const { editor } = this;
+		const viewDocument = editor.editing.view.document;
+
+		this.listenTo<ViewDocumentTouchStartEvent>( viewDocument, 'touchstart', ( evt, data ) => {
```

Comment:
This should be tighter coupled with the above click-at-the-end of link handling so both handlers would not fight for the same actions.